test/transparentproxy/install/install.go

```diff
@@ -0,0 +1,396 @@
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/exec"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/kumahq/kuma/pkg/test/matchers"
+	. "github.com/kumahq/kuma/test/framework"
+	"github.com/kumahq/kuma/test/transparentproxy/utils"
+)
+
+var (
+	// Generic
+	cmdUserAdd = []string{"useradd", "-u", "5678", "kuma-dp"}
+	// Debian, Ubuntu
+	cmdAptUpdate          = []string{"apt-get", "update", "-y"}
+	cmdAptInstallIptables = []string{"apt-get", "install", "-y", "iptables"}
+	// RHEL, Amazon Linux
+	cmdYumUpdate          = []string{"yum", "update", "-y"}
+	cmdYumInstallIptables = []string{"yum", "install", "-y", "iptables"}
+	// On Amazon Linux, the `useradd` command is not available by default.
+	// The `shadow-utils` package, which provides user management tools,
+	// includes `useradd`.
+	cmdYumInstallShadowUtils = []string{"yum", "install", "-y", "shadow-utils"}
+	// Alpine
+	cmdAddUser        = []string{"adduser", "-u", "5678", "kuma-dp", "-D"}
+	cmdApkAddIptables = []string{"apk", "add", "iptables"}
+)
+
+type testCase struct {
+	name            string
+	image           string
+	postStart       [][]string
+	additionalFlags []string
+}
+
+func Install() {
+	DescribeTable(
+		"kumactl install transparent-proxy inside Docker container",
+		func(tc testCase) {
+			Expect(TProxyConfig.KumactlLinuxBin).NotTo(BeEmpty())
+
+			container, err := testcontainers.GenericContainer(
+				context.Background(),
+				testcontainers.GenericContainerRequest{
+					ContainerRequest: testcontainers.ContainerRequest{
+						Image:      tc.image,
+						Privileged: true,
+						Files: []testcontainers.ContainerFile{{
+							HostFilePath:      TProxyConfig.KumactlLinuxBin,
+							ContainerFilePath: "/usr/local/bin/kumactl",
+							FileMode:          0o700,
+						}},
+						Cmd: []string{"sleep", "infinity"},
+						LifecycleHooks: []testcontainers.ContainerLifecycleHooks{
+							{PostStarts: utils.BuildContainerHooks(tc.postStart)},
+						},
+						WaitingFor: wait.ForExec([]string{"kumactl", "version"}).
+							WithStartupTimeout(time.Second * 10).
+							WithExitCodeMatcher(func(exitCode int) bool {
+								return exitCode == 0
+							}).
+							WithResponseMatcher(func(body io.Reader) bool {
+								data, _ := io.ReadAll(body)
+								return bytes.Contains(data, []byte("Client: "))
+							}),
+					},
+					Started: true,
+				},
+			)
+			Expect(err).ToNot(HaveOccurred())
+
+			DeferCleanup(func() {
+				Expect(container.Terminate(context.Background())).To(Succeed())
+			})
+
+			EnsureInstallSuccessful(container, tc.additionalFlags)
+			EnsureGoldenFiles(container, tc)
+		},
+		EntriesForImages(TProxyConfig.DockerImagesToTest),
+		XEntriesForImages(TProxyConfig.DockerImagesToTestPaused),
+	)
+}
+
+func EnsureInstallSuccessful(container testcontainers.Container, flags []string) {
```

Comment:
I think that `EnsureInstallSuccessful` and `EnsureGoldenFiles` could be moved to the `framework` directory.

Reply:
I don't think so to be honest as they are specific to these particular tests (install transparent proxy) and in current form there is extremely low probability these functions could be reused anywhere else.